homeassistant/components/slack/__init__.py

```diff
@@ -40,7 +42,8 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     slack = WebClient(token=entry.data[CONF_API_KEY], run_async=True, session=session)
 
     try:
-        await slack.auth_test()
+        slack.url = (await slack.auth_test())["url"]
+        slack.user_id = (await slack.auth_test())["user_id"]
```

Comment:
This is strange: we're awaiting the same coroutine twice and using its returned data to update properties of the same object that carries the coroutine?

Reply:
oops